lib/cli/src/split_version.rs

```diff
@@ -0,0 +1,726 @@
+            InvalidPackageName(p) => write!(f, "parsing as package: {p}"),
+            FileDoesNotExist(path) => write!(f, "parsing as file: file does not exist: {path}"),
+            InvalidDirectory(path) => write!(
+                f,
+                "parsing as file: file is a directory, but contains no wapm.toml: {path}"
+            ),
+        }
+    }
+}
+
+/// Error(s) that can happen when parsing a `SplitVersion`
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct SplitVersionMultiError {
+    /// Original string that was parsed, used to fixup wasmer run args
+    pub original: String,
+    /// Errors that happen when trying to resolve the package
+    ///
+    /// e.g.: error 1: tried to access filesystem, but file isn't there
+    ///       error 2: tried to resolve package, but package doesn't exist
+    ///       error 3: tried to find command, but command doesn't exist
+    pub errors: Vec<SplitVersionError>,
+}
+
+impl fmt::Display for SplitVersionMultiError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "\r\n")?;
+        for e in self.errors.iter() {
+            write!(f, "    {}\r\n", e)?;
+        }
+        Ok(())
+    }
+}
+
+impl std::error::Error for SplitVersionMultiError {}
+
+impl SplitVersion {
+    /// Parse a version from a String
+    pub fn parse(s: &str) -> Result<SplitVersion, SplitVersionMultiError> {
+        s.parse()
+    }
+}
+
+impl FromStr for SplitVersion {
+    type Err = SplitVersionMultiError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut errors = Vec::new();
+
+        match SplitVersionInner::try_parse_file(s) {
+            Ok(o) => {
+                return Ok(Self {
+                    original: s.to_string(),
+                    inner: o,
+                })
+            }
+            Err(e) => {
+                errors.push(e);
+            }
+        }
+
+        match SplitVersionInner::try_parse_url(s) {
+            Ok(o) => {
+                return Ok(Self {
+                    original: s.to_string(),
+                    inner: o,
+                })
+            }
+            Err(e) => {
+                errors.push(e);
+            }
+        }
+
+        match SplitVersionInner::try_parse_package(s) {
+            Ok(o) => {
+                return Ok(Self {
+                    original: s.to_string(),
+                    inner: o,
+                })
+            }
+            Err(e) => {
+                errors.push(e);
+            }
+        }
+
+        match SplitVersionInner::try_parse_command(s) {
+            Ok(o) => {
+                return Ok(Self {
+                    original: s.to_string(),
+                    inner: o,
+                })
+            }
+            Err(e) => {
+                errors.push(e);
+            }
+        }
+
+        Err(SplitVersionMultiError {
+            original: s.to_string(),
+            errors,
+        })
```

Comment:
I don't really think it makes sense to collect multiple errors here. 
Should just return something like "InvalidPackageSpecifier" here, that's enough.

Reply:
No, this is on purpose, so we can print something like:

```sh
wasmer run qjsj.temp
error: Invalid value "qjsj.temp" for '<FILE>': 
    parsing as file: file does not exist: qjsj.temp
    parsing as URL: relative URL without a base
    parsing as package: Invalid package: "qjsj.temp"
    parsing as command: invalid command: qjsj.temp


For more information try --help
```

This way it's much easier to debug if something fails. There are cases where you can't distinguish between a package and a filepath, so if we error on one case, this might cause problems when debugging such cases.

Reply:
I don't think that's very helpful for the user, rather just confusing.

I'd just print a generic error message like "error: 

Invalid run target "qjsj.temp".
You can specify a local path, run packages from the Wasmer registry with ............................